weave/scorers/coherence_scorer.py

```diff
@@ -0,0 +1,107 @@
+from typing import Optional, Union
+
+from pydantic import Field, PrivateAttr, validate_call
+
+import weave
+from weave.scorers.default_models import MODEL_PATHS
+from weave.scorers.scorer_types import HuggingFacePipelineScorer
+from weave.scorers.utils import (
+    WeaveScorerResult,
+    ensure_hf_imports,
+    load_hf_model_weights,
+)
+
+
+class WeaveCoherenceScorerV1(HuggingFacePipelineScorer):
+    """
+    The scorer that assesses if the model output is coherent using a fine-tuned
+    deberta-small-long-nli model from tasksource, https://huggingface.co/tasksource/deberta-small-long-nli
+    Use wandb/coherence_scorer to check if the model output is coherent.
+    Args:
+        model_name: The name of the coherence scorer model to use. Defaults to `wandb/coherence_scorer`.
+        device: The device to use for inference. Defaults to `auto`, which will use `cuda` if available.
+        model_max_length: The maximum length of the model output. Defaults to 1024.
+    """
+
+    task: str = "sentiment-analysis"
+    model_max_length: int = Field(
+        default=1024, description="The maximum length of the model output."
+    )
+    _label2id: dict[str, int] = PrivateAttr(
+        default_factory=lambda: {
+            "Completely Incoherent": 0,
+            "Mostly Incoherent": 1,
+            "A Little Incoherent": 2,
+            "Mostly Coherent": 3,
+            "Perfectly Coherent": 4,
+        }
+    )
```

Comment:
shouldn't this just be a constant in the file?  Why does it need to be on the scorer at all?

Reply:
I agree, but don't think it is =(. I will try to move this to the scorer itself as it an in-house one

Reply:
I think you can just put it in this file as `label2id = {...}`  It doesn't need to be part of the scorer

Reply:
Ok, repacked them !